data_visualization_in_open_source_software/data_visualization.md

```diff
@@ -0,0 +1,265 @@
+
+```python
+import seaborn as sns
+```
+
+The seaborn functions are designed to work well with [pandas](https://pandas.pydata.org/) data structures, which are organized a little differently (and generally much better for data science purposes!) from the default data structures in python. If you're not using pandas dataframes, though, [you can still use seaborn](https://seaborn.pydata.org/tutorial/data_structure.html).
+
+Because seaborn is built on top of matplotlib, it doesn't automatically show plots you create, it waits until you ask to see them. If you want to see the plots you're making, you need to either use the command `matplotlib.pyplot.show()` after each plot or, if you're using an interactive jupyter notebook you can put it in [matplotlib mode](https://ipython.readthedocs.io/en/stable/interactive/plotting.html).
+
+## How plotting works in practice
+
+Often, the best way to get the visualization you want is by finding an existing example that's similar to what you want and then copying that code and modifying it as needed.
+
+<div class = "important">
+Don't expect to become fully "fluent" in either ggplot2 or seaborn! They are complex and powerful systems, with lots and lots of detail. Even experienced programmers frequently rely on google to get their plotting code just right.
+</div>
+
+### Selecting the right plot for your data
+
+There are many kinds of plots available, and in most cases more than one kind of plot would be a reasonable choice for your data.
+
+When selecting a plot type, consider
+
+- the type(s) of variables you wish to plot (especially whether they are continuous or categorical)
+- what idea or story you want your plot to communicate
+
+When you're just getting started with data visualization, it's a good idea to spend as much time as you can examining plots you think work well --- What is it that makes them effective and attractive? How do they communicate information? What can you tell about the underlying data from examining the plot?
+
+Google's Material Design provides a [good overview of most kinds of plots and their typical uses](https://material.io/design/communication/data-visualization.html#selecting-charts). Browsing through that collection and others can help give you ideas of plots to try.
+
+### Data visualization workflow
+
+Here is a good workflow for creating a new data visualization using ggplot2 or seaborn:
+
+1. Spend time carefully [thinking through the plot you want to create](#selecting-the-right-plot-for-your-data). For many people, this means drawing a sketch (a great excuse to keep colored pencils at your desk!). Include as many details as you can, like labels, facets, and scales.
```

Comment:
I was a bit surprised when I clicked this link and it took me back to the page I had just come from -- as a learner, I think I would have expected a link to an external webpage. 

I wonder if a better way to reference this may be to just say "...thinking through the plot you want to create using the considerations discussed on the previous page" or something to that effect?